File: android/src/main/java/ru/vvdev/yamap/suggest/MapSuggestClient.java
package ru.vvdev.yamap.suggest;

import com.facebook.react.bridge.ReadableMap;
import com.yandex.mapkit.search.SuggestOptions;

import java.util.List;

import ru.vvdev.yamap.utils.Callback;

public interface MapSuggestClient {
    /**
     * Получить саджесты по тексту {@code text}.
     * Вернуть результат в метод {@code onSuccess} в случае успеха, в случае неудачи в {@code onError}
     */
    void suggest(final String text, final Callback<List<MapSuggestItem>> onSuccess, final Callback<Throwable> onError);
    void suggest(final String text, final ReadableMap options, final Callback<List<MapSuggestItem>> onSuccess, final Callback<Throwable> onError);
    /**
     * Остановить сессию поиска саджестов
     */
    void resetSuggest();
}

File: android/src/mapkit/full/suggest/MapSuggestClient.kt
package ru.yamap.suggest

import com.facebook.react.bridge.ReadableMap
import ru.yamap.utils.Callback

interface MapSuggestClient {
    /**
     * Получить саджесты по тексту `text`.
     * Вернуть результат в метод `onSuccess` в случае успеха, в случае неудачи в `onError`
     */
    fun suggest(
        text: String?,
        onSuccess: Callback<List<MapSuggestItem?>?>?,
        onError: Callback<Throwable?>?
    )

    fun suggest(
        text: String?,
        options: ReadableMap?,
        onSuccess: Callback<List<MapSuggestItem?>?>?,
        onError: Callback<Throwable?>?
    )
    /**
     * Остановить сессию поиска саджестов
     */
    fun resetSuggest()
}
